ddr/ddr/Models/GameEntities/Bat.swift

```diff
@@ -83,41 +92,16 @@ class Bat: SKSpriteNode {
   
   // annoying but required - doing the minimum to compile
   required init?(coder aDecoder: NSCoder) {
-    flapping = nil
-    audioManager = AudioManager()
-    z = Bat.maxZMagnitude
-    super.init(coder: aDecoder)
-  }
-  
-  func move() {
-    z += velocity
-  }
-  
-  func isGone() -> Bool {
-    return abs(z) >= Bat.maxZMagnitude
-  }
-  
-  func die() {
-    flapping?.stop()
-    removeFromParent()
+    fatalError("init(coder:) has not been implemented")
```

Comment:
Stick it to the man!

...This won't cause problems, will it?

Reply:
I don't think so, but at the very least it is better (in my opinion) than what we had before because the old code would fail silently and likely have bizarre problems whereas this will fail in a spectacular and impossible to miss fashion :)